### PR TITLE
Enable support for building rz-libswift on cutter

### PR DESCRIPTION
**Pull request by wargio**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://cutter.re/docs/contributing/code/getting-started.html) to this repository
- [x] I made sure to follow the project's [coding style](https://cutter.re/docs/contributing/code/development-guidelines.html)
- [ ] I've updated the [documentation](https://cutter.re/docs/user-docs.html) with the relevant information (if needed)

**Detailed description**

Adds building support for rz-libswift on cutter.

(build code in cutter is untested)

### CODE REVIEW

**Comment by wargio** [comment]
build is broken because requires Rizin v0.4.0
I'm going to patch this once we have a new version of cutter

**Comment by XVilka** [comment]
You can update the `rizin` submodule.

File: dist/bundle_rz_libswift.ps1 [hunk 1]
```diff
@@ -0,0 +1,16 @@
+$dist = $args[0]
+$python = Split-Path((Get-Command python.exe).Path)
+
+if (-not (Test-Path -Path 'libswift' -PathType Container)) {
+    git clone https://github.com/rizinorg/rz-libswift.git --depth 1 libswift
+}
+cd libswift
+& meson.exe --buildtype=release -Drizin_plugdir=lib/plugins --prefix=$dist --libdir=lib/plugins --datadir=lib/plugins build
```
Comment (ret2libc) [inline dist/bundle_rz_libswift.ps1]:
```suggestion
& meson.exe --buildtype=release --prefix=$dist build
```
I think that is not necessary, is it?

File: scripts/rz-libswift.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+cd "$SCRIPTPATH/.."
+
+if [[ ! -d libswift ]]; then
+	git clone https://github.com/rizinorg/rz-libswift.git --depth 1 libswift
+fi
+
+cd libswift
+rm -rf build || sleep 0
+mkdir build && cd build
+meson --buildtype=release -Drizin_plugdir=share/rizin/plugins --libdir=share/rizin/plugins --datadir=share/rizin/plugins "$@" ..
```
Comment (ret2libc) [inline scripts/rz-libswift.sh]:
```suggestion
meson --buildtype=release "$@" ..
```

same

File: dist/CMakeLists.txt [hunk 3]
```diff
@@ -93,6 +105,17 @@ if(APPLE)
 		")
 	endif()
 
+	if (CUTTER_PACKAGE_RZ_LIBSWIFT AND CUTTER_ENABLE_DEPENDENCY_DOWNLOADS)
+		install(CODE "
+			execute_process(COMMAND \"${CMAKE_CURRENT_SOURCE_DIR}/../scripts/rz-libswift.sh\" --prefix=\${CMAKE_INSTALL_PREFIX}
```
Comment (ret2libc) [inline dist/CMakeLists.txt]: add ` --pkg-config-path=\${CMAKE_INSTALL_PREFIX}/lib/pkgconfig` similarly to what's done few lines above for jsdec.

**Comment by wargio** [comment]
On linux it does work, checked via `bins/mach0/swift-main`

image

**Comment by wargio** [comment]
@thestr4ng3r @xarkes

**Pull request merged**